File: PHP/8php 1/fake_bin.php
<?php
/*
|--------------------------------------------------------------------------
| Fake Binary
|--------------------------------------------------------------------------
|
*/

function fake_bin(string $s): string {
   $res = '';
   for ($i=0; $i < strlen($s); $i++) { 
      $res .= ((int)$s[$i]) < 5 ? "0" : "1";
   }
   return $res;
}


function fake_bin(string $s): string {
   return strtr($s, '0123456789', '0000011111');
}


function fake_bin(string $s): string {
   return preg_replace(['/[1-4]/','/[5-9]/'], [0,1], $s);
}


function fake_bin(string $s): string {
   $s = str_replace(range(0,4), 0, $s);
   $s = str_replace(range(5,9), 1, $s);
   return $s;
}


print_r(fake_bin('45385593107843568'). ' 01011110001100111
');
print_r(fake_bin('509321967506747'). ' 101000111101101
');
print_r(fake_bin('366058562030849490134388085'). ' 011011110000101010000011011
');

File: Python/8py 1/fake_bin.py
"""
|--------------------------------------------------------------------------
| Fake Binary
|--------------------------------------------------------------------------
|
"""

def fake_bin(x):
   res = []
   for i in range(len(x)):
      res.append('0') if int(x[i]) < 5 else res.append('1')
   return ''.join(res)


def fake_bin(x):
   return ''.join('0' if c < '5' else '1' for c in x)


import string
def fake_bin(s):
   return s.translate(string.maketrans('0123456789', '0000011111'))


def fake_bin(x):
   return ''.join([str(int(i) // 5) for i in x])


def fake_bin(x):
   map = str.maketrans('123456789', '000011111')
   return x.translate(map)


print(fake_bin('45385593107843568'), '01011110001100111')
print(fake_bin('509321967506747'), '101000111101101')
print(fake_bin('366058562030849490134388085'), '011011110000101010000011011')